boefjes/tests/test_app.py

```diff
@@ -193,6 +196,100 @@ def test_null(manager: SchedulerWorkerManager, tmp_path: Path, item_handler: Moc
     }
 
 
+def test_one_process_deduplication_of_tasks(manager: SchedulerWorkerManager, item_handler: MockHandler, tmp_path):
+    manager.scheduler_client = MockSchedulerClient(
+        boefje_responses=[get_dummy_data("scheduler/pop_response_duplicated_boefje.json")],
+        normalizer_responses=[],
+        log_path=tmp_path / "patch_task_log",
+    )
+    with pytest.raises(KeyboardInterrupt):
+        manager.run(WorkerManager.Queue.BOEFJES)
+
+    items = item_handler.get_all()
+
+    # Just one task dispatched
+    assert len(items) == 1
+    assert items[0].boefje.id == "dns-records"
+
+    patched_tasks = manager.scheduler_client.get_all_patched_tasks()
+
+    # But two tasks marked as completed
+    assert len(patched_tasks) == 3
+    assert set(patched_tasks) == {
+        ("70da7d4f-f41f-4940-901b-d98a92e9014b", "running"),
+        ("70da7d4f-f41f-4940-901b-d98a92e9014b", "completed"),
+        ("a0da7d4f-f41f-4940-901b-d98a92e9014b", "completed"),
+    }
+
+    bytes_calls = item_handler.bytes_client.get_all()
+    assert bytes_calls == [
+        (
+            "save_boefje_meta",
+            (
+                {
+                    "id": UUID("a0da7d4f-f41f-4940-901b-d98a92e9014b"),
+                    "started_at": None,
+                    "ended_at": None,
+                    "boefje": {"id": "dns-records", "version": None},
+                    "input_ooi": "Hostname|internet|test.test",
+                    "arguments": {},
+                    "organization": "_dev2",
+                    "runnable_hash": None,
+                    "environment": None,
+                },
+            ),
+        ),
+        ("save_raw", (UUID("a0da7d4f-f41f-4940-901b-d98a92e9014b"), b"123", {"boefje/dns-records", "my/mime"})),
+    ]
+
+
+def test_one_process_deduplication_exception_puts_duplicated_task_back_on_the_queue(
+    manager: SchedulerWorkerManager, item_handler: MockHandler, tmp_path
+):
+    manager.scheduler_client = MockSchedulerClient(
+        boefje_responses=[get_dummy_data("scheduler/pop_response_duplicated_boefje_error.json")],
+        normalizer_responses=[],
+        log_path=tmp_path / "patch_task_log",
+    )
+    with pytest.raises(KeyboardInterrupt):
+        manager.run(WorkerManager.Queue.BOEFJES)
+
+    items = item_handler.get_all()
+    assert len(items) == 0
```

Comment:
```suggestion
    assert not items
```

Reply:
A bit inconsistent with the other tests here though 🤔

Reply:
Well it's consistent with PEP (using implicit booleanness to validate an empty list is quite Pythonic) and the codebase.

I've also looked into the other tests in this module and it seems that the emptiness checks are already inconsistent in style unfortunately